### PR TITLE
A-MQ xPaaS docs after firs QE review.

### PR DESCRIPTION
**Pull request by ppenicka**
Hi Bilhar, submitting a pull request for the A-MQ xPaaS docs, with the quick-fix comments from QE implemented.

### CODE REVIEW

**Comment by adellape** [comment]
cc @CowboysFan

File: using_images/xpaas_images/a_mq.adoc [hunk 1]
```diff
@@ -5,4 +5,92 @@
 :icons:
 :experimental:
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Can you update the metadata and other bits to match the latest topic template here?

https://raw.githubusercontent.com/openshift/openshift-docs/master/contributing_to_docs/templates/topic_template.adoc

Of note, adding `:prewrap!` as well as including a TOC (`toc::[]`) and Overview section.

**Comment by adellape** [comment]
@ppenicka While they're already pretty close, I feel like there is some opportunity to sync up some of the style/structure (minor stuff) a little more between this and the other two existing xPaaS image topics. If you and @lucascosti are already talking about that sorta stuff, then don't mind me!

File: using_images/xpaas_images/a_mq.adoc [hunk 2]
```diff
@@ -5,4 +5,92 @@
...
 
-Information on the Red Hat JBoss A-MQ xPaaS image will be added soon.
+Red Hat JBoss A-MQ (JBoss A-MQ) is available as a containerized xPaaS image that is designed for use with OpenShift v3. It allows developers to quickly deploy an A-MQ message broker in a hybrid cloud environment.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Suggest dropping "v3" here, and/or conditionalize it between "OpenShift Enterprise 3.0" and <something> for Origin.

File: using_images/xpaas_images/a_mq.adoc [hunk 3]
```diff
@@ -5,4 +5,92 @@
...
+* Configuration of the broker can be performed:
+** using parameters specified in the A-MQ application template, as described in xref:configuring-params[Application Template Parameters].
+** using the STI (Source-to-image) tool, as described in xref:configuring-sti[Configuration Using STI].
+
+
+== Installing the JBoss A-MQ xPaaS Images
+
+To use the Red Hat xPaaS middleware images in your OpenShift project, you must first install the image streams and source-to-image (S2I) application templates.
+
+. In a terminal, change into the root directory of your OpenShift master. It should contain the 'examples' subdirectory.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: These instructions as they're currently written assume the reader is an admin with access to the master host. A typical OpenShift user would likely not. For comparison, the other two xPaaS image topics[1][2] link here to the admin "First Steps" topic (though they could be more explicit that they're being linked to an admin-task) which details this process (for OSE Ansible-based installations, these should be imported by default anyway).

Was it intentional to detail these steps for this image here?

cc @bparees for thoughts.

[1] https://docs.openshift.com/enterprise/3.0/using_images/xpaas_images/eap.html#installing-the-jboss-eap-xpaas-image-streams-and-application-templates
[2] https://docs.openshift.com/enterprise/3.0/using_images/xpaas_images/jws.html#installing-the-jboss-web-server-xpaas-image-streams-and-application-templates

File: using_images/xpaas_images/a_mq.adoc [hunk 4]
```diff
@@ -5,4 +5,92 @@
...
+. Create the S2I templates for your A-MQ application:
++
+----
+# oc create -f examples/application-templates/amq/amq6.json -n <project>
+# oc create -f examples/application-templates/amq/amq6-persistence.json -n <project>
+----
+
+You will now have the JBoss A-MQ xPaaS images stream and S2I application templates available for use in your OpenShift project. 
+
+To create an application based on the template, follow instructions in the *Creating a Configuration From a Template* section of the link:../dev_guide/templates.html[Creating New Applications] topic in the OpenShift documentation. For a reference on template parameters specified when creating the application, see the xref:configuring-params[Application Template Parameters] section below.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Bad link:

```
link:../dev_guide/templates.html[Creating New Applications]
```

Should go up one more level.

Also, the topic title and section seem to be off or out of date. Were you meaning to link to `new_app.html#specifying-a-template`? I.e, the "Specifying a Template" section of the "Creating New Applications" topic. Or was it meant to be `templates.html#creating-resources-from-a-template` for the "Creating Resources from a Template" section of the "Templates" topic?

File: using_images/xpaas_images/a_mq.adoc [hunk 5]
```diff
@@ -5,4 +5,92 @@
...
+
+[[configuring-params]]
+=== Application Template Parameters
+
+Basic configuration of the JBoss A-MQ xPaaS image is performed by specifying values of application template parameters. The following parameters can be configured:
+
+AMQ_RELEASE::
+  The JBoss A-MQ release version. This determines which JBoss A-MQ image will be used as a basis for the application. At the moment, only version _6.2_ is available.
+APPLICATION_NAME::
+  The name of the application used internally in OpenShift. It is used in names of services, pods and other objects within the application.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Comma: "services, pod, and other objects"

File: using_images/xpaas_images/a_mq.adoc [hunk 6]
```diff
@@ -5,4 +5,92 @@
...
+MQ_USERNAME::
+  The user name used for authentication to the broker. In a standard non-containerized JBoss A-MQ, you would specify the user name in the 'AMQ_HOME/opt/user.properties' file. If no value is specified, a random user name is generated.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Markup for file names/directories:

```
*_AMQ_HOME/opt/user.properties_*
```

See: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference

File: using_images/xpaas_images/a_mq.adoc [hunk 7]
```diff
@@ -5,4 +5,92 @@
...
+MQ_PASSWORD::
+  The password used for authentication to the broker. In a standard non-containerized JBoss A-MQ, you would specify the password in the 'AMQ_HOME/opt/user.properties' file. If no value is specified, a random password is generated.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Same re: file name markup.

File: using_images/xpaas_images/a_mq.adoc [hunk 8]
```diff
@@ -5,4 +5,92 @@
...
+  Comma-separated list of the messaging protocols used by the broker. Available options are _amqp_, _mqtt_, _openwire_, and _stomp_. If left empty, all available protocols will be available. Please note that for integration of the image with Red Hat JBoss Enterprise Application Platform, the _openwire_ protocol must be specified, while other protocols can be optionally specified as well.
+MQ_QUEUES::
+  Comma-separated list of queues available by default on the broker on its startup.
+MQ_TOPICS::
+  Comma-separated list of topics available by default on the broker on its startup.
+
+[[configuring-sti]]
+=== Configuration Using STI
+
+Configuration of the JBoss A-MQ image can also be modified using the Source-to-image feature, described in full detail at link:../s2i_images/overview.html[Source-to-image (S2I)].
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: I would suggest linking here to either `../../creating_images/s2i.html` or `../../architecture/core_concepts/builds_and_image_streams.adoc#source-build`, which have more information about S2I.

File: using_images/xpaas_images/a_mq.adoc [hunk 9]
```diff
@@ -5,4 +5,92 @@
...
+
+Custom A-MQ broker configuration can be specified by creating an 'openshift-activemq.xml' file inside the 'git' directory of your application's Git project root. On each commit, the file will be copied to the 'conf' directory in the A-MQ root and its contents used to configure the broker.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: File name/dir markup.

File: using_images/xpaas_images/a_mq.adoc (hunk 3 above)
Comment (bparees) [inline using_images/xpaas_images/a_mq.adoc]: yeah this should link to the first steps doc for now, though ultimately this whole situation w/ how templates get installed and updated is going to need to change.

File: using_images/xpaas_images/a_mq.adoc [hunk 10]
```diff
@@ -5,4 +5,92 @@
...
+
+== Security
+
+Only SSL connections can connect from outside of the OpenShift instance, regardless of the protocol specified in the _MQ_PROTOCOL_ property of the A-MQ application templates. The non-SSL version of the protocols can only be used inside the OpenShift instance.
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Markup for config params / env vars:

```
`*MQ_PROTOCOL*`
```

File: using_images/xpaas_images/a_mq.adoc [hunk 11]
```diff
@@ -5,4 +5,92 @@
...
+== Scalability
+
+It is possible to run multiple JBoss A-MQ containers on a single OpenShift instance at the same time. However, real-time scalability is not supported as messages stored on a discarded broker would be lost when downscaling.
+
+== Logging
+
+In addition to viewing the OpenShift logs, you can troubleshoot a running JBoss A-MQ image by viewing the JBoss A-MQ logs that are outputted to the container's console:
+
+----
+oc logs -f <pod_name> <container_name>
```
Comment (adellape) [inline using_images/xpaas_images/a_mq.adoc]: Slap a `$` on the beginning?

**Comment by ppenicka** [comment]
Hi Alex,

Thanks for the thorough review, all comments reflected in the latest commit.

Also my apologies for doing a pull request from master, but looking at the diff on the 'Files changed' tab, this should not have any side effects as I've used master just for this single task.

**Comment by adellape** [comment]
@ppenicka Changes look good, thanks! Merged via https://github.com/openshift/openshift-docs/pull/757.

**Pull request closed without merging**